upup/pkg/fi/cloudup/awstasks/iamoidcprovider.go

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awstasks
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"k8s.io/klog"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup/awsup"
+	"k8s.io/kops/upup/pkg/fi/cloudup/cloudformation"
+	"k8s.io/kops/upup/pkg/fi/cloudup/terraform"
+)
+
+//go:generate fitask -type=IAMOIDCProvider
+type IAMOIDCProvider struct {
+	Lifecycle *fi.Lifecycle
+
+	ARN         *string
+	ClientIDs   []*string
+	Thumbprints []*string
```

Comment:
FYI in a later PR I changed this to be `[]fi.Resource`, so we can defer evaluation

Reply:
I left this as-is so we can keep your commit that changes it, since your commit also touches some other packages and I'd like to keep this PR small in scope

Reply:
:+1: